controllers/external/util.go

```diff
@@ -152,6 +152,19 @@ func FailuresFrom(obj *unstructured.Unstructured) (string, string, error) {
 	return failureReason, failureMessage, nil
 }
 
+// FailureDomain returns the FailureDomain from the external object spec.
+func FailureDomain(obj *unstructured.Unstructured) *string {
+	result, found, err := unstructured.NestedFieldNoCopy(obj.Object, "spec", "failureDomain")
```

Comment:
Can you use `NestedString` here?

Reply:
likely, foggy brain from a crazy day. Will give it a go tomorrow after I've rested a bit :)